colossalai/inference/config.py

```diff
@@ -340,3 +351,23 @@ def from_dict(cls, config_dict: Dict[str, Any]) -> "InferenceConfig":
         # Set the attributes from the parsed arguments.
         inference_config = cls(**inference_config_args)
         return inference_config
+
+
+@dataclass
+class ModelInferenceConfig:
+    """
+    Configurations used when initializing/sharding model for inference.
+
+    Args:
+        dtype (torch.dtype): The data type for weights and activations.
+        use_cuda_kernel (bool): Whether to use cuda kernel, faster but lose some precision occasionally
+        use_spec_dec (bool): Indicate whether to use speculative decoding.
+        use_flash_attn (bool): Indicate whether to use flash attention.
+        use_cuda_graph (bool): Whether to enforce CUDA graph execution. If False, we will disable CUDA graph and always execute the model in eager mode. If True, we will use eager execution in hybrid.
+    """
+
+    dtype: torch.dtype = None
+    use_cuda_kernel: bool = False
+    use_spec_dec: bool = False
+    use_flash_attn: bool = False
+    use_cuda_graph: bool = False
```

Comment:
`use_cuda_graph` should not be categorized into model inference config but an engine feature

Reply:
Removed.